client/client-multi/client-c/src/platform-impl/kaa_tcp_channel.c

```diff
@@ -988,7 +1062,18 @@ kaa_error_t kaa_tcp_channel_authorize(kaa_tcp_channel_t *self)
                                                            , &sync_buffer
                                                            , &sync_size);
 
-    if (error_code) {
+    KAA_LOG_INFO(self->logger, error_code, "Kaa TCP channel [0x%08X] going to send CONNECT message (%zu bytes)"
+                                                                                , self->access_point.id, sync_size);
+
+    kaa_error_t delete_error_code = kaa_tcp_channel_delete_pending_services(self
+                                                       , self->pending_request_services
+                                                       , self->pending_request_service_count);
+    if (delete_error_code) {
+        KAA_LOG_ERROR(self->logger, delete_error_code, "Kaa TCP channel [0x%08X] failed to delete pending services"
+                                                                                , self->access_point.id);
+    }
+
+    if (error_code || delete_error_code) {
```

Comment:
I'e have changed code, now we return correct error code in both cases if error_code or delete_error_code is set. And kaa_tcp_channel user will receive error on event processing.
But I'm agree that we need to notify SDK if channel can't pack sync message in any case.